pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/model/Matrix.groovy

```diff
@@ -40,7 +40,7 @@ import org.jenkinsci.plugins.scriptsecurity.sandbox.whitelists.Whitelisted
 class Matrix extends Parallel {
 
     @Whitelisted
-    Matrix(List<Stage> stages) {
```

Comment:
Since there is a limit on ListExpression size, we pass a list of lists and then concatenate.